packages/jamf-adapter/src/definitions/fetch/transforms/utils.ts

```diff
@@ -79,3 +91,27 @@ export const removeSelfServiceSecurityPassword = (value: Record<string, unknown>
     }
   }
 }
+
+export const maskPayloadsPassword = (value: Values): void => {
+  const payloads = _.get(value, 'general.payloads')
+  if (typeof payloads === 'string') {
+    if (PASSWORD_REGEX.test(payloads)) {
+      log.trace(`Masked value of payloads in '${_.get(value, 'general.name')}'`)
+      _.set(value, 'general.payloads', payloads.replace(PASSWORD_REGEX, MASKED_PASSWORD))
```

Comment:
We're now testing the regex twice which isn't ideal for long strings.
I think you can provide `replace` with a callback, so you can log once you detect the regex. should be something like -
```suggestion
const maskedPayloads = payloads.replace(PASSWORD_REGEX, () => {
    log.trace(`Masked value of ...`)
    return MASKED_PASSWORD
  })
  _.set(value, 'general.payloads', maskedPayloads)
```

if it doesn't work probably best to just skip the log 😅

Reply:
yep, actually I had mixed feelings regarding this log initially, even after seeing we have such a log in Jira. But I added it as your request became a tie breaker 😅
Looks like it could work with _replace_ but not with _replaceAll_, and I don't want to postpone the rollout of this, will revisit it while working on the CV 🙏